ext/fs/fat/zfs_diskio.c

```diff
@@ -29,13 +29,15 @@
 #include <ffconf.h>
 #include <disk_access.h>
 
+static const char* const pdrv_str[] = {_VOLUME_STRS};
+
 /*-----------------------------------------------------------------------*/
 /* Get Drive Status                                                      */
 /*-----------------------------------------------------------------------*/
 
 DSTATUS disk_status(BYTE pdrv)
 {
-	if (disk_access_status() != 0) {
+	if (disk_access_status(pdrv_str[pdrv]) != 0) {
```

Comment:
Please add an assertion here to ensure that pdrv is within the bounds of pdrv_str.  While it'll be removed in release mode, it provides some guidance to perform validations in syscalls later.

Reply:
Ok